### PR TITLE
[hotfix/hybridengine] fix bug when tp*pp size = 1

### PR DESCRIPTION
**Pull request by FoolPlayer**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

fix bug when tp*pp size = 1

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Review by tiandiao123 — APPROVED** [review]

File: colossalai/inference/engine/engine.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def __init__(
         # Init pg mesh
         pg_mesh = ProcessGroupMesh(pp_size, tp_size)
 
-        stage_manager = PipelineStageManager(pg_mesh, PP_AXIS, True)
+        stage_manager = PipelineStageManager(pg_mesh, PP_AXIS, True if pp_size * tp_size > 1 else False)
```
Comment (ver217) [inline colossalai/inference/engine/engine.py:129]: `PipelineStageManager` is not related to tp. Why consider tp here?

File: colossalai/inference/engine/engine.py [hunk 2]
```diff
@@ -142,7 +142,9 @@ def __init__(
         self.verbose = verbose
         self.schedule = GenerateSchedule(stage_manager, self.mb_manager, verbose)
 
-        self.model = self._shardformer(model, model_policy, stage_manager, pg_mesh.get_group_along_axis(TP_AXIS))
+        self.model = self._shardformer(
+            model, model_policy, stage_manager, pg_mesh.get_group_along_axis(TP_AXIS) if pp_size * tp_size > 1 else None
```
Comment (ver217) [inline colossalai/inference/engine/engine.py:146]: Same. This is not related to PP

**Review by ver217 — APPROVED** [review]

File: colossalai/inference/engine/engine.py (hunk 1 above)
Comment (FoolPlayer) [reply colossalai/inference/engine/engine.py:129]: Cause hybridengine use pp and tp together with the same pipeline logic, which is also worked when pipeline size is 1.
